### PR TITLE
Set DOCKER_HOST for kind in vscode container

### PR DESCRIPTION
**Pull request by giusdp**
This PR closes issue #71 in nuvolaris/nuvolaris

### CODE REVIEW

File: nuv/kind.go [hunk 1]
```diff
@@ -29,6 +30,9 @@ type KindCmd struct {
 
 func Kind(args ...string) error {
 	os.Args = append([]string{"kind"}, args...)
+	if err := patchDockerHost(); err != nil {
+		fmt.Printf("WARNING: docker host patching failed: %s! If you're inside a container kind won't work.\n", err.Error())
```
Comment (msciabarra) [inline nuv/kind.go:34]: Here if you "patched" docker host just return

File: nuv/util.go [hunk 2]
```diff
@@ -127,3 +128,41 @@ func WriteFileToNuvolarisConfigDir(filename string, content []byte) (string, err
...
+		return err
+	}
+	// and is a socket
+	if info.Mode().Type() != fs.ModeSocket {
+		return fmt.Errorf("docker-host.sock exists but it's not a socket")
+	}
+
+	// and $DOCKER_HOST is empty
+	if len(os.Getenv("DOCKER_HOST")) > 0 {
+		return fmt.Errorf("$DOCKER_HOST not empty")
```
Comment (msciabarra) [inline nuv/util.go:157]: Here is you have DOCKER_HOST set it means you are the "forked" version so just return a nil so you can move forward with kind

File: nuv/util.go [hunk 3]
```diff
@@ -127,3 +128,41 @@ func WriteFileToNuvolarisConfigDir(filename string, content []byte) (string, err
...
+	}
+
+	// then sudo env DOCKER_HOST=unix:///var/run/docker-host.sock nuv devcluster"
+	// os.Setenv("DOCKER_HOST", "unix:///var/run/docker-host.sock")
+	cmd := exec.Command("sudo", "env", "DOCKER_HOST=unix:///var/run/docker-host.sock")
```
Comment (msciabarra) [inline nuv/util.go:162]: here you should just re-execute sudo env DOCKER_HOST and append os.Args() so it re-execute itself under sudo with the env variable changed

File: nuv/kind.go [hunk 4]
```diff
@@ -18,6 +18,7 @@
 package main
```
Comment (msciabarra) [inline nuv/kind.go:18]: The basic idea is:

if <I am in a container>
  if DOCKER_HOST is set
     exit
  else
      sudo env DOCKER_HOST=.... + os.Args()

**Review by irinacodes — APPROVED** [review]

**Pull request merged**